slick.core.js

```diff
@@ -675,10 +721,239 @@
   function regexSanitizer(dirtyHtml) {
      return dirtyHtml.replace(/(\b)(on[a-z]+)(\s*)=|javascript:([^>]*)[^>]*|(<\s*)(\/*)script([<>]*).*(<\s*)(\/*)script(>*)|(&lt;)(\/*)(script|script defer)(.*)(&gt;|&gt;">)/gi, '');
   }
- 
-  // exports
-  $.extend(true, window, {
-    "Slick": {
+
+  // With help from https://youmightnotneedjquery.com/
+  function grep( elems, callback, invert ) {
+		var callbackInverse,
+			matches = [],
+			i = 0,
+			length = elems.length,
+			callbackExpect = !invert;
+
+		// Go through the array, only saving the items
+		// that pass the validator function
+		for ( ; i < length; i++ ) {
+			callbackInverse = !callback( elems[ i ], i );
+			if ( callbackInverse !== callbackExpect ) {
+				matches.push( elems[ i ] );
+			}
+		}
+
+		return matches;
+	}
+
+  function offset(el) {
+
+    box = el.getBoundingClientRect();
+    docElem = document.documentElement;
+    return {
+      top: box.top + window.pageYOffset - docElem.clientTop,
+      left: box.left + window.pageXOffset - docElem.clientLeft
+    };
+ 	}
+
+  function width(el, value) {
+    if (value === undefined) {
+      return el.getBoundingClientRect().width;
+    }
+    setStyleSize(el, "width", value);
+  }
+
+  function height(el, value) {
+    if (value === undefined) {
+      return el.getBoundingClientRect().height;
+    }
+    setStyleSize(el, "height", value);
+  }
+
+  function setStyleSize(el, style, val) {
+    if (typeof val === 'function') val = val();
+    if (typeof val === 'string') el.style[style] = val;
+    else el.style[style] = val + 'px';
+  }
+
+  function contains(parent, child) {
+    if(!parent || !child)
+      return false;
+
+    const parentList = parents(child);
+    return !parentList.every((p) => {
```

Comment:
hmm up to this point we were still supporting the ES5 syntax and not yet syntax like `() => {}`, we might need to change this to keep compatibility since we aren't transpiling our code. Note that there a few more of these in this file too

Reply:
Done